test/remote/gateways/remote_blue_snap_test.rb

```diff
@@ -52,6 +52,20 @@ def test_successful_purchase
     assert_equal 'Success', response.message
   end
 
+  def test_successful_fractionless_currency_purchase
+    options = @options.merge(currency: 'BYR')
+    response = @gateway.purchase(1234, @credit_card, options)
+    assert_success response
+    assert_equal 'Success', response.message
+  end
```

Comment:
I printed the response on this and got: `..."amount"=>"0.00", "usd-amount"=>"0.00", "currency"=>"USD"...`, so I bumped the amount to `123400` on this test and got `Transaction failed  because of payment processing failure.: 1000 - There are no E-Commerce merchant accounts configured to process Visa transactions with currency 974.` I'm guessing BlueSnap does not fully support the `BYR` currency? Instead of `BYR`, maybe we can use `JPY` (as it's a more widely used non-fractional currency)? We'll get a more consistent response from the gateway. Also, maybe we should test with an amount that ends in two zeros (since they'll be dropped off in the request anyway)?

Reply:
Great catch! Thank you for running that locally. I changed it over to JPY and tested locally and it is now behaving as we would expect.